azure/services/groups/spec.go

```diff
@@ -55,7 +56,7 @@ func (s *GroupSpec) Parameters(ctx context.Context, existing *asoresourcesv1.Res
 			Tags: infrav1.Build(infrav1.BuildParams{
 				ClusterName: s.ClusterName,
 				Lifecycle:   infrav1.ResourceLifecycleOwned,
-				Name:        ptr.To(s.Name),
+				Name:        ptr.To(azure.GetNormalizedKubernetesName(s.Name)),
```

Comment:
If this change isn't necessary to fix the bug, I'd revert it just to reduce the amount of potential churn here.